2.18 What's new and Language evolution pages (#4195)
dart-lang/site-www: src/_guides/whats-new.md

This section lists notable changes made from May 12, 2022,
through August 30, 2022.
For details about the 2.18 release,
see [Dart 2.18: Objective-C & Swift interop][],

> and the [change log][].
> 
> [Dart 2.18: Objective-C & Swift interop]: https://medium.com/dartlang/dart-2-18-f4b3101f146c
> [change log]: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#2180

Reviewer: I think we generally don't put a space in "changelog" and perhaps we should be more specific what changelog this is referring to:

```suggestion
and the [SDK changelog][].

[Dart 2.18: Objective-C & Swift interop]: https://medium.com/dartlang/dart-2-18-f4b3101f146c
[SDK changelog]: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#2180
```

In addition to bug fixes and incremental improvements,
we made the following changes to this site:

> * Added a workaround to [Fixing common type problems][], 
>   for the rare case where type inference might still incorrectly infer an argument type is null.

Reviewer 2: **suggestion:** Simplify the copy and try not to use the same word twice in the same sentence. Also, changed to active voice.

```suggestion
* Added a workaround to [Fixing common type problems][]. 
  This corrects for when type inference might be conclude that an argument type can use a null value.
```

* Removed all mention of discontinued `.packages` files from [What not to commit][].
  If you still need to generate a `.packages` file due to third-party legacy depedencies,
  see [`dart pub get` Options][].
* Removed dedicated pages and any other mention of discontinued `dart2js` and `dartdevc` command line tools.

>   * Added command line options and known issues previously associated with `dart2js` to the [`dart compile`]() page.

Reviewer: When separating the link definition, use `[]` after the link text:

```suggestion
  * Added command line options and known issues previously associated with `dart2js` to the [`dart compile`][] page.
```

>   * Added information on debugging production code to [Debugging Dart web apps]().

Reviewer: ```suggestion
  * Added information on debugging production code to [Debugging Dart web apps][].
```

This section lists notable changes made from May 12, 2022,
through August 30, 2022.
For details about the 2.18 release,
see [Dart 2.18: Objective-C & Swift interop][],
and the [change log][].

[Dart 2.18: Objective-C & Swift interop]: https://medium.com/dartlang/dart-2-18-f4b3101f146c
[change log]: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#2180

In addition to bug fixes and incremental improvements,
we made the following changes to this site:

* Added a workaround to [Fixing common type problems][], 
  for the rare case where type inference may still incorrectly infer an argument type is null.
* Removed all mention of discontinued `.packages` files from [What not to commit][].
  * See [][] if you still need to generate a `.packages` file due to third-party legacy depedencies.
* Adjsuted the [dart2js compiler][] page and the [dart command line tool][] page to reflect the full removal of the previously deprecated `dart2js` command.
  * Other command line tools discontinued in this release,
  `dartdevc` and `dartanalyzer`,
  were already removed from the documentation when deprecated.

> * Updated the [Asynchronous programming: futures, async, await][] codelab to reflect subtle behavior changes from the revamped methods.

Reviewer: The first note in [List or table](https://developers.google.com/style/lists?hl=en#list-or-table).

> * Updated the [Asynchronous programming: futures, async, await][] codelab to reflect subtle behavior changes from the revamped methods.

Reviewer: I'm also curious about the subtle behavior changes/revamped methods mention. Is it expected for the async-await codelab to need some changes due to the async-await changes in Dart 2.18?

> * Updated the [Asynchronous programming: futures, async, await][] codelab to reflect subtle behavior changes from the revamped methods.

Author: Aha... I saw a post-2.17 pr open related to async, so assumed it was related to the changes mentioned in the change log. Thought I got off easy on that one haha.

I'll cross-check the descriptions in the change log with these  async docs that seem to be related: https://dart.dev/guides/language/concurrency and 
https://dart.dev/guides/libraries/futures-error-handling, 
to see if anything actually needs to be updated.

> Is it expected for the async-await codelab to need some changes due to the async-await changes in Dart 2.18?

I'm not capable of making that call atm.
